### PR TITLE
testing/percona-xtradb-cluster: new aport

### PR DESCRIPTION
**Pull request by Flowman**
https://www.percona.com/software/mysql-database/percona-xtradb-cluster
Percona XtraDB Cluster is an active/active high availability and high
scalability open source solution for MySQL clustering

### CODE REVIEW

**Comment by Flowman** [comment]
This failed because the Travis logfile got to big. Any suggestions?

**Comment by jirutka** [comment]
Don't worry about it, we'll verify it locally.

**Comment by Flowman** [comment]
Don't test this one yet as i found an issue in wsrep_sst_xtrabackup-v2. It uses the wrong timeout command.

Testing a fix for it and will commit a change.

**Comment by Flowman** [comment]
Another question. I noticed in my testing now that /usr/bin/wsrep_sst_xtrabackup-v2 requires socat and iproute2(ss) when used with percona-xtrabackup. Should I add it as a dependency or does the end-user have to figure that one out?

File: testing/percona-xtradb-cluster/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+                -DINSTALL_SUPPORTFILESDIR=share/mysql \
+                -DINSTALL_MYSQLSHAREDIR=share/mysql \
+                -DINSTALL_DOCDIR=share/mysql/docs \
+                -DINSTALL_SHAREDIR=share/mysql \
+                -DMYSQL_UNIX_ADDR=/run/mysqld/mysqld.sock \
+                -DDEFAULT_CHARSET=utf8 \
+                -DDEFAULT_COLLATION=utf8_general_ci \
+                -DMYSQL_SERVER_SUFFIX="-$WSREP_VERSION" \
+                -DWITH_INNODB_DISALLOW_WRITES=ON \
+                -DDOWNLOAD_BOOST=1 \
```
Comment (fabled) [inline testing/percona-xtradb-cluster/APKBUILD]: Alpine policy is to use system provided libraries; not to bundle them. There are few exceptions, but we'd rather not add more things.

File: testing/percona-xtradb-cluster/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+                -DWITH_BOOST="$builddir"/libboost \
+                -DWITH_WSREP=ON \
+                -DWITH_UNIT_TESTS=0 \
```
Comment (fabled) [inline testing/percona-xtradb-cluster/APKBUILD]: We now have `check()` that should be used for running unit tests.

File: testing/percona-xtradb-cluster/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+                -DWITH_READLINE=system \
+                -DWITHOUT_TOKUDB=ON \
+                -DCOMPILATION_COMMENT="$COMMENT" \
+                -DWITH_PAM=OFF \
+                -DWITH_INNODB_MEMCACHED=ON \
+                -DWITH_SCALABILITY_METRICS=ON \
+                || return 1
```
Comment (fabled) [inline testing/percona-xtradb-cluster/APKBUILD]: `|| return 1` no longer needed at all as abuild runs the functions with `set -e`

File: testing/percona-xtradb-cluster/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+        mysqld_safe mysqltest perror replace resolve_stack_dump resolveip
+        wsrep_sst_common wsrep_sst_mysqldump wsrep_sst_xtrabackup-v2 wsrep_sst_rsync
+        mysql_ssl_rsa_setup"
+    mkdir -p "$subpkgdir"/usr/bin/ || return 1
+    for i in $bins; do
+        mv "$pkgdir"/usr/bin/${i} "$subpkgdir"/usr/bin/ || return 1
+    done
+}
+
+md5sums="58f8524be1e25cc740cfa009d3e7c09c  Percona-XtraDB-Cluster-5.7.16-27.19.tar.gz
```
Comment (fabled) [inline testing/percona-xtradb-cluster/APKBUILD]: Do `abuild checksum` with new abuild as `md5sums` and `sha256sums` are no longer needed or generated.

File: testing/percona-xtradb-cluster/fix-posix_timers.patch [hunk 5]
```diff
@@ -0,0 +1,50 @@
+--- a/mysys/posix_timers.c
++++ b/mysys/posix_timers.c
+@@ -188,10 +188,20 @@
+           On error, -1 is returned, and errno is set to indicate the error.
+ */
+
++struct linux_sigevent {
```
Comment (fabled) [inline testing/percona-xtradb-cluster/fix-posix_timers.patch]: This is horribly wrong hack. See PR #808 review comments on how to fix this properly.

**Comment by clandmeter** [comment]
Closed due to inactivity.

**Pull request closed without merging**